pkg/advisory/data_session.go

```diff
@@ -51,10 +51,15 @@ func NewDataSession(ctx context.Context, opts DataSessionOptions) (*DataSession,
 
 	ds.githubClient = opts.GitHubClient
 
+	gitAuth, err := wgit.GetGitAuth(opts.Distro.Absolute.AdvisoriesHTTPSCloneURL())
+	if err != nil {
+		return nil, fmt.Errorf("getting git auth: %w", err)
+	}
+
 	// clone advisories repo
 	repo, err := git.PlainCloneContext(ctx, tempDir, false, &git.CloneOptions{
 		URL:  opts.Distro.Absolute.AdvisoriesHTTPSCloneURL(),
-		Auth: wgit.GetGitAuth(),
+		Auth: gitAuth,
```

Comment:
nit: If we save the value of `opts.Distro.Absolute.AdvisoriesHTTPSCloneURL()` to a local variable, we can use that variable down below in the clone options instead of requesting the URL a second time. I like that because it helps us keep the _token_ and the _URL it goes to_ in-sync as we make further changes to the code.

Reply:
👍